test/cqlpy/test_secondary_index.py

```diff
@@ -1956,3 +1956,16 @@ def test_paging_and_drop_index_no_allow_filtering(cql, test_keyspace):
                 assert len(r.current_rows) <= page_size
                 got.extend(r.current_rows)
             assert expected == got
+
+
+# Test index representation in system.* tables
+def test_index_in_system_tables(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "p int PRIMARY KEY, v int") as table:
+        index_name = unique_name()
+        cql.execute(f"CREATE INDEX {index_name} ON {table}(v)")
+        res = [ f'{r.keyspace_name}.{r.view_name}' for r in cql.execute('select * from system.built_views')]
+        assert f'{test_keyspace}.{index_name}_index' in res
+        res = [ f'{r.table_name}::{r.index_name}' for r in cql.execute('select * from system."IndexInfo"')]
+        assert f'{test_keyspace}::{index_name}' in res
+        res = [ f'{r.table_name}::{r.index_name}' for r in cql.execute(f'select * from system."IndexInfo" where table_name = \'{test_keyspace}\' AND index_name = \'{index_name}\'')]
+        assert f'{test_keyspace}::{index_name}' in res
```

Comment:
Nitpick: If I understand correctly the result of this select is a single row, so you can just assert the value **equals** res, not that it's **in** res.

Reply:
Not quite, this test can share scylla instance with some other test, so if this "other" test plays with indexes/views, selection from system.IndexInfo table would report extra lines, so the check is `in`

Reply:
My question was about the **second** select request, where you explicitly select with test_keyspace and index_name. That will always return just one result - no matter which other indexes exist.

For the first select, you're obviously right that the "in" is needed.

Reply:
Ah, right. Good point